File: Sad.cpp
#include <iostream>

class Emotion {
private:
  bool imSad;

public:
  Emotion() { this->imSad = true; }
  bool sad() {
    std::cout << "When I get sad, ";
    return this->imSad;
  }
  void sadStop() {
    this->imSad = false;
    std::cout << "I stop being sad ";
  }
  void beAwesome() { std::cout << "and be awesome instead"; }
};

int main(void) {
  Emotion *myEmotion = new Emotion();

  // life motto
  if (myEmotion->sad() == true) {
    myEmotion->sadStop();
    myEmotion->beAwesome();
  }
}

File: Sad.ts
class Emotion {
  #imSad: boolean;
  constructor() {
    this.#imSad = true;
  }
  sad(): boolean {
    process.stdout.write("When I get sad, ");
    return this.#imSad;
  }
  sadStop(): void {
    this.#imSad = false;
    process.stdout.write("I stop being sad ");
  }
  beAwesome(): void {
    process.stdout.write("and be awesome instead");
  }
}

const myEmotion: Emotion = new Emotion();

// life motto
if (myEmotion.sad() === true) {
  myEmotion.sadStop();
  myEmotion.beAwesome();
}

